Documentation/alternate-fetches.md

```diff
@@ -0,0 +1,87 @@
+</body>                                                                         
+<script>                                                                        
+    const idp = "https://solidcommunity.net";                                   
+    const privateResource = "https://jeff-zucker.solidcommunity.net/private/";  
+    const auth = solidClientAuthentication;                                     
+    async function main(session){                                               
+        const kb = $rdf.graph();                                                
+        window.solidFetcher = session.fetch;                                    
+        const fetcher = $rdf.fetcher(kb)                                        
+        try {                                                                   
+          await fetcher.load(privateResource);                                  
+          alert("Private resource successfully loaded");                        
+        }                                                                       
+        catch(e) { alert(e) }                                                   
+    }                                                                           
+    document.getElementById('login').onclick = ()=> {                           
+        auth.login({                                                            
+            oidcIssuer: idp,                                                    
+            redirectUrl: window.location.href,                                  
+            clientName: "rdflib test"                                           
+        });                                                                     
+    }                                                                           
+    async function handleRedirectAfterLogin() {                                 
+        await auth.handleIncomingRedirect();                                    
+        session = auth.getDefaultSession();                                     
+        if (session.info.isLoggedIn)  main(session);                            
+    }                                                                           
+    handleRedirectAfterLogin();                                                 
+</script></html>                                                                
+```
+
+
+##  Custom fetchers
+
+Another way to specify an authenticated or alternate fetch is to do it when you create the fetcher object.  This method follows the same steps as the first method, with two exceptions: omit the line *`global.solidFetch = auth.fetch`*; and instead of the line *`const fetcher = $rdf.fetcher(kb)`* use :
+```javascript
+const fetcher = $rdf.fetcher(kb,{fetch:auth.fetch.bind(auth)});
+```
+Doing things this way avoids using global variables, but also means that the changes to fetch need to be added every time you create a new fetcher object.
```

Comment:
Is this true ?
My understanding was that the fetch was installed as $rdf.fetch so was kept if you do not create a new rdf object. It is the way it is used in SolidOS. You can create new store $rdf.graph() or $rdf.fetcher() and keep the same $rdf.fetch.

Reply:
@bourgeoa - I believe you are wrong.  I tested and could not make it work.  If you create a fetcher with fetcher = $rdf.fetcher(kb) without options it will be an unauthenticted fetch even if you previously created a fetcher that was authenticated.  If you have a test case where the authentication is preserved across creation of fetchers, please post it.

Reply:
This succeeds on the first fetch and fails 401 on the second : 
```
const auth = new (require("solid-node-client").SolidNodeClient)();              
const $rdf = global.$rdf = require('rdflib');                                   
const privateResource = "https://jeff-zucker.solidcommunity.net/private/";      
async function test() {                                                         
    await auth.login();                                                         
    const kb = $rdf.graph();                                                    
    const fetcher1 = $rdf.fetcher(kb,{fetch:auth.fetch.bind(auth)});            
    try {                                                                       
       await fetcher1.load(privateResource);                                    
       console.log("Private resource successfully loaded");                     
     }                                                                          
    catch(e) { console.log(e) }                                                 
    const fetcher2 = $rdf.fetcher(kb);                                          
    try {                                                                       
       await fetcher2.load(privateResource);                                    
       console.log("Private resource successfully loaded");                     
     }                                                                          
    catch(e) { console.log(e) }                                                 
}                                                                               
test();    
```

Reply:
If I am correct in this, it is really important that within solidOS, if we need to create a new fetcher, we grab the _fetch from the existing fetch method in case the user has over-ridden it.

Reply:
Yes you are right. I was fully wrong. Fetcher do not default to current fetch.

Reply:
Worth checking, though, thanks for your careful eye.